pkg/providers/cloudstack/validator_registry_test.go

```diff
@@ -0,0 +1,36 @@
+package cloudstack_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/providers/cloudstack"
+	"github.com/aws/eks-anywhere/pkg/providers/cloudstack/decoder"
+)
+
+func getFromRegistry(registry cloudstack.ValidatorRegistry, execConfig *decoder.CloudStackExecConfig) (*cloudstack.Validator, error) {
+	return registry.Get(execConfig)
+}
+
+func TestRegistryGetWithNilExecConfig(t *testing.T) {
+	g := NewGomegaWithT(t)
+	registry := cloudstack.NewValidatorFactory(executables.NewLocalExecutablesBuilder(), nil, false)
+	_, err := getFromRegistry(registry, nil)
+	g.Expect(err).NotTo(BeNil())
+}
+
+func TestRegistryGetSuccess(t *testing.T) {
+	g := NewGomegaWithT(t)
+	registry := cloudstack.NewValidatorFactory(executables.NewLocalExecutablesBuilder(), nil, false)
+	_, err := getFromRegistry(registry, &decoder.CloudStackExecConfig{Profiles: []decoder.CloudStackProfileConfig{
```

Comment:
can we check the returned value it not nil and it's a valid validator struct?

Reply:
I'm not sure how we could check that it's a valid validator struct, but we can definitely check not nil

Reply:
If this was returning an interface, which I think it should, just making a type assertion

Reply:
Yeah, I think it should return an interface. If not, you can't mock it in the controller and you are back to mocking the underlying client.

Reply:
Done. But the linter complained when I added the type assertion, saying 
```
pkg/providers/cloudstack/validator_registry_test.go:36:11: S1040: type assertion to the same type: validator already has type cloudstack.ProviderValidator (gosimple)
	_, ok := validator.(cloudstack.ProviderValidator)
	         ^
```
so I removed it.